c/tskit/trees.c

```diff
@@ -3565,6 +3565,30 @@ tsk_tree_get_sites(tsk_tree_t *self, tsk_site_t **sites, tsk_size_t *sites_lengt
     return 0;
 }
 
+int TSK_WARN_UNUSED
+tsk_tree_depth(tsk_tree_t *self, tsk_id_t u, unsigned int *depth_ret)
+{
+    tsk_id_t parent;
+    unsigned int depth;
+    int ret = 0;
+
+    ret = tsk_tree_check_node(self, u);
+    if (ret != 0) {
+        goto out;
+    }
+
+    parent = self->parent[u];
```

Comment:
A bit shorter:
```c
depth = 0; 
for (v = self->parent[u]; v != TSK_NULL; v = self->parent[v]) {
    depth++;
}
```
I think ``parent`` is a slightly misleading variable name as it's a iteration variable and using ``v`` for it is a bit more idiomatic. I don't feel strongly about it though, no biggie.